mongodump/mongodump_test.go

```diff
@@ -872,6 +872,94 @@ func TestMongoDumpTOOLS2174(t *testing.T) {
 	})
 }
 
+// Test dumping a collection while respecting no index scan for wired tiger.
+func TestMongoDumpTOOLS1952(t *testing.T) {
+	testtype.SkipUnlessTestType(t, testtype.IntegrationTestType)
+	log.SetWriter(ioutil.Discard)
+
+	sessionProvider, _, err := testutil.GetBareSessionProvider()
+	if err != nil {
+		t.Fatalf("No cluster available: %v", err)
+	}
+
+	session, err := sessionProvider.GetSession()
+	if err != nil {
+		t.Fatalf("Failed to get session: %v", err)
+	}
+
+	collName := "tools-1952-dump"
+	dbName := "test"
+	ns := dbName + "." + collName
+
+	var r1 bson.M
+
+	dbStruct := session.Database(dbName)
+
+	sessionProvider.Run(bson.D{{"drop", collName}}, &r1, dbName)
+
+	createCmd := bson.D{
+		{"create", collName},
+	}
+	var r2 bson.M
+	err = sessionProvider.Run(createCmd, &r2, dbName)
+	if err != nil {
+		t.Fatalf("Error creating collection: %v", err)
+	}
+
+	// Check whether we are using MMAPV1.
+	isMMAPV1 := db.IsMMAPV1(dbStruct, collName)
+
+	// Turn on profiling.
+	profileCmd := bson.D{
+		{"profile", 2},
+	}
+
+	err = sessionProvider.Run(profileCmd, &r2, dbName)
+	if err != nil {
+		t.Fatalf("Failed to turn on profiling: %v", err)
+	}
+
+	profileCollection := dbStruct.Collection("system.profile")
+
+	Convey("testing dumping a collection query hints", t, func() {
+		md := simpleMongoDumpInstance()
+		md.ToolOptions.Namespace.Collection = collName
+		md.ToolOptions.Namespace.DB = dbName
+		md.OutputOptions.Out = "dump"
+		err = md.Init()
+		So(err, ShouldBeNil)
+		err = md.Dump()
+		So(err, ShouldBeNil)
+
+		c, err := profileCollection.Find(context.Background(),
+		bson.D{
+			{"ns", ns},
+			{"op", "query"},
+			{"$or", []interface{}{
+				// 4.0+
+				bson.D{{"command.hint._id", 1}},
+				// 3.6
+				bson.D{{"command.$nsapshot", true}},
+				bson.D{{"command.snapshot", true}},
+				// 3.4 and previous
+				bson.D{{"query.$snapshot", true}},
+				bson.D{{"query.snapshot", true}},
+				bson.D{{"query.hint._id", 1}},
+			}},
+		},
+		)
+		So(err, ShouldBeNil)
+		count := testutil.CountCursorResults(c)
+		if isMMAPV1 {
+			// There should be exactly one query that matches.
+			So(count, ShouldEqual, 1)
```

Comment:
Does this assertion actually ever get hit in our tests? I thought I remembered you mentioning that none of the tests actually run with an MMAPv1 MongoDB instance

Reply:
Right, but I've tested the queries manually. I think we should add some MMAPv1 tests whenever we get to the test coverage epic!

Reply:
By "queries" I mean using export and dump on mmapv1 mongod instance manually and inspecting the generated queries.

Reply:
I'm fine with deferring to a testing epic if it's been tested manually! Can you file a ticket for that and add it to the appropriate epic?

Reply:
Sure thing!